ucm2/alcatel-idol3/HiFi.conf

```diff
@@ -0,0 +1,57 @@
+# Use case configuration for Alcatel OneTouch Idol 3.
+# The two speakers/earpieces are connected to external codec (TFA9897)
+# Headphones support requires writing a driver for AK4375
+
+Define {
+	WcdCapturePCM "hw:${CardId},0"
+}
+<platforms/msm8916/qdsp6-components.conf>
+
+SectionDevice."Speaker" {
+	Comment "Speaker playback"
+
+	EnableSequence [
+		cset "name='Speaker Top Amp Input' Left"
+		cset "name='Speaker Bottom Amp Input' Right"
+		cset "name='Speaker Top Amp Switch' 1"
+		cset "name='Speaker Bottom Amp Switch' 1"
+		cset "name='Speaker Top Mode' Speaker"
+		cset "name='Speaker Bottom Mode' Speaker"
+	]
+
+	DisableSequence [
+		cset "name='Speaker Top Amp Switch' 0"
+		cset "name='Speaker Bottom Amp Switch' 0"
+	]
+
+	Value {
+		PlaybackChannels 2
+		PlaybackPriority 300
+		PlaybackPCM "hw:${CardId},1"
+	}
+}
+
+SectionDevice."Earpiece" {
+	Comment "Earpiece playback"
+
+	EnableSequence [
+		cset "name='Speaker Top Amp Input' Left"
+		cset "name='Speaker Top Amp Switch' 1"
+		cset "name='Speaker Top Mode' Receiver"
```

Comment:
the mode switch should happen before the Switch in case there is some audio playing at the moment (otherwise it would switch while playing)